### ISSUE TITLE
Field with Enum type does not respect schema override

### ISSUE THREAD

**Issue by hellozyemlya** [samuelcolvin/pydantic#1856]
# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
D:\Develop\pydantic>python -c "import pydantic.utils; print(pydantic.utils.version_info())"
             pydantic version: 1.6.1
            pydantic compiled: False
                 install path: D:\Develop\pydantic\pydantic
               python version: 3.8.3 (tags/v3.8.3:6f8c832, May 13 2020, 22:37:02) [MSC v.1924 64 bit (AMD64)]
                     platform: Windows-10-10.0.19041-SP0
     optional deps. installed: []

```

```py
from enum import Enum
from typing import Optional

from pydantic import BaseModel, Field


class Operation(Enum):
    """
    Enum that represents comparison operations.
    """
    MORE = ">"
    LESS = "<"


class Test(BaseModel):
    operation: Operation = Field(default=Operation.LESS,
                                 description="Operation used to compare something with something.")


print(Test.schema_json(indent=2))
"""
prints:

{
  "title": "Test",
  "type": "object",
  "properties": {
    "operation": {
      "$ref": "#/definitions/Operation"
    }
  },
  "definitions": {
    "Operation": {
      "title": "Operation",
      "description": "Enum that represents comparison operations.",
      "enum": [
        ">",
        "<"
      ]
    }
  }
}
"""
```
Its expected to have default value and description alongside with $ref.

Fix is pretty straightforward:
```diff
diff --git a/pydantic/schema.py b/pydantic/schema.py
index 27c66b2..ee8b331 100644
--- a/pydantic/schema.py
+++ b/pydantic/schema.py
@@ -702,6 +702,10 @@ def field_singleton_schema(  # noqa: C901 (ignore complexity)
         enum_name = normalize_name(field_type.__name__)
         f_schema = {'$ref': ref_prefix + enum_name}
         definitions[enum_name] = enum_process_schema(field_type)
+        if not schema_overrides:
+            return f_schema, definitions, nested_models
+        else:
+            return {'allOf': [f_schema]}, definitions, nested_models
     else:
         add_field_type_to_schema(field_type, f_schema)

```

**Comment by PrettyWood** [comment]
Hello @hellozyemlya 
Fix is already done here #1749 👍

**Comment by hellozyemlya** [comment]
> Hello @hellozyemlya
> Fix is already done here #1749 👍

Nice. Waiting for it to be committed.
